src/main/scala/org/broadinstitute/dsde/workbench/sam/service/TosService.scala

```diff
@@ -0,0 +1,31 @@
+package org.broadinstitute.dsde.workbench.sam.service
+
+import cats.effect.IO
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.workbench.model.{WorkbenchEmail, WorkbenchGroupName}
+import org.broadinstitute.dsde.workbench.sam.dataAccess.DirectoryDAO
+import org.broadinstitute.dsde.workbench.sam.model.BasicWorkbenchGroup
+import org.broadinstitute.dsde.workbench.sam.util.SamRequestContext
+
+class TosService (val directoryDao: DirectoryDAO) extends LazyLogging {
+
+  def createNewGroupIfNeeded(currentVersion:Int, isEnabled:Boolean): IO[Any] = {
+    if(!isEnabled || groupExists(currentVersion)){
+      IO.unit
+    } else {
+      logger.info("creating new ToS group")
+      directoryDao.createGroup(BasicWorkbenchGroup(WorkbenchGroupName(getGroupName(currentVersion)),
+        Set.empty, WorkbenchEmail("foo@bar.com")), samRequestContext = SamRequestContext(None))
```

Comment:
Looking for suggestions about what to do with the email for the new group.  Should we move that to config?  Will the email be different per environment?

Reply:
You could do something similar to [how the All_Users group does it.](https://github.com/broadinstitute/sam/blob/develop/src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala#L80) The email address _domain_ will be unique per env (because we've got a different subdomain for each env), but the prefix won't be. And the domain is already in config.

All_Users is probably the closest analog we've got to these ToS groups.